kratos/python_scripts/read_materials_process.py

```diff
@@ -64,32 +64,33 @@ def _GetVariable(self,my_string):
 
         return KratosMultiphysics.KratosGlobals.GetVariable(splitted[-1]) # This also checks if the application has been imported
 
-    def _GetConstitutiveLaw(self,my_string):
+    def _GetConstitutiveLaw(self, my_string):
         """Return the python object of a Constitutive Law named by the string argument.
 
         Example:
         constitutive_law = self._GetConstitutiveLaw('KratosMultiphysics.StructuralMechanicsApplication.LinearElastic3DLaw')
         model_part.GetProperties(prop_id).SetValue(CONSTITUTIVE_LAW, constitutive_law)
         """
         splitted = my_string.split(".")
+
         if(len(splitted) == 0):
-            raise Exception("something wrong. Trying to split the string "+my_string)
+            raise Exception("something wrong. Trying to split the string " + my_string)
         if(len(splitted) == 1):
-            raise Exception("Please also provide the name of the application of constitutive law "+my_string)
+            raise Exception("Please also provide the name of the application of constitutive law " + my_string)
+        if len(splitted) > 3:
+            raise Exception("Something wrong. String " + my_string + " has too many arguments")
 
         constitutive_law_name = splitted[-1]
-        module_name = splitted[-2]
+        application_name = splitted[-2]
 
-        if module_name == "KratosMultiphysics":
+        if application_name == "KratosMultiphysics":
             return getattr(KratosMultiphysics, constitutive_law_name) 
         else:
-            application_name = "Kratos" + module_name
-            if application_name not in KratosMultiphysics.KratosGlobals.RequestedApplications:
-                raise ImportError(module_name + " is not imported!")
-            module1 = KratosMultiphysics.KratosGlobals.RequestedApplications[application_name]
-            module2 = sys.modules[application_name]
+            # Check that application was imported in the main script
+            KratosMultiphysics.CheckRegisteredApplications(application_name)
+            application = __import__("Kratos" + application_name)
```

Comment:
shouldn't this be "KratosMultiphysics" + application_name ?

Reply:
I don’t think so, usually it is „Kratos“ plus the name of the application, see here: [StructuralMechanics](https://github.com/KratosMultiphysics/Kratos/blob/8351413e20f80278168e7db065924e5c76cc236c/applications/StructuralMechanicsApplication/StructuralMechanicsApplication.py#L3) , [FluidDynamics](https://github.com/KratosMultiphysics/Kratos/blob/8351413e20f80278168e7db065924e5c76cc236c/applications/FluidDynamicsApplication/FluidDynamicsApplication.py#L4)
Also I havent changed this, I only changed the name of the variable to make it more clear

Reply:
but for example we write

import KratosMultiphysics.StructuralMechanicsApplication as StructuralMechanicsApplication

Reply:
That's true
I just tried it with your proposal but that didn't work.
I checked and the names of my shared libs are always `KratosXYApplication.so`, e.g. `KratosStructuralMechanicsApplication.so`
I think this is why it works with what I did before.
The problem with `__import__(KratosMultiphysics.StructuralMechanicsApplication)` is that it returns `KratosMultiphysics` and not `StructuralMechanicsApplication`

I think we now have three(four) options:
1. `application = getattr(KratosMultiphysics, application_name)`
2. `application = __import__("Kratos" + application_name)` 
3. `application = importlib.import_module("KratosMultiphysics." + application_name)` => I think this only works from python 2.7 onwards. On the other hand this seems to be the new thing and also you had used it in an earlier version of this process
4. (`__import__("KratosMultiphysics", fromlist=[application_name])` => not sure if this works and seems hacky)

As a conclusion I vote for 1 (Since I check beforehand if the app has been imported) or 3
Your call

Reply:
Importing `"Kratos"+application_name` imports the underlying `.so` library (so the C++ parts only). It is not importing the python module `KratosMultiphysics.ApplicationName`, which is a small difference but (if I remember correctly) means that the `python_scripts` folder for that application will not be visible. This may or may not matter to you, but I thought I'd mention it. It also does not do the registering and so on, but this should not be as important here, since you do this only after checking that the application was imported.